### DISCUSSION TITLE
Bind Route Info to Component Inputs (✨ New Router feature) - justangular.com

### DISCUSSION THREAD

**Discussion by eneajaho** [eneajaho/just-angular#7] [General]
https://justangular.com/blog/bind-route-info-to-component-inputs-new-router-feature

**Comment by michaelwatkins** [comment]
Thank you for the article.  Do you know of a way to bind the route to signal inputs?

**Reply by eneajaho** [reply]
> Thank you for the article.  Do you know of a way to bind the route to signal inputs?

Hi, it should work the same way.

**Comment by simeyla** [comment]
Interestingly with signals if you have both a signal input and a @Input() in the same file, only the signal gets given the value! So if you migrate to signals you need to remove @Input().

  @Input('type')
  typeInput!: string;

  typeSignal = input('type');

**Comment by simeyla** [comment]
As an alternative to renaming Input on import, I'm doing this :

  queryParam_type = input('type');

Wish I could do this, but it's not allowed because the compiler isn't clever enough:

   queryString = { 
      type: input('type'),
      color: input('color')
   }

**Comment by simeyla** [comment]
Correction: Some of my examples needed an alias - since the first parameter is the initial value and not the property name. I knew that ;-) 

queryParam_type = input(null, { alias: 'type' })

**Comment by bennadel** [comment]
Thanks for the in-depth write-up. One thing that we used to be able to do in the earlier days of Angular 2+ was bind inputs to sub-structures of a component using the `inputs` component meta-data:

```js
@Component({
    inputs: {
        "sub.structure.id": "id" // Map "id" to the "this.sub.structure.id"
    }
})
```

It seems that this is no longer possible (trying it in Angular 18).  But, looking at your writeup, I've got an idea to move this logic into the Route resolve. I'm not exactly sure what it will look like, just thinking out loud.
